plugins/inputs/logfile/tail/tail_queue_test.go

```diff
@@ -0,0 +1,27 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package tail
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestTailerEnqueueAndDequeue(t *testing.T) {
+	tailerEnqueue := NewTailerFifoQueue()
+	tailerEnqueue.Enqueue()
+	tailerEnqueue.Enqueue()
+	assert.Equal(t, 2, tailerEnqueue.Size())
+
+	ok := tailerEnqueue.Dequeue()
```

Comment:
Again, I strongly disagree with this because this is misusing a queue. We don't care what the value is. We care about the size and capacity of the counter.

Reply:
Sorry I'm still in the middle of transitioning and have not planned to touch this in the morning.